api/services/SiteCreator.js

```diff
@@ -82,6 +82,7 @@ const createSiteFromExistingRepo = ({ siteParams, user }) => {
 const createSiteFromTemplate = ({ siteParams, user, template }) => {
   let site = Site.build(siteParams)
   site.engine = "jekyll"
+  site.defaultBranch = templateForTemplateName(template).branch
```

Comment:
Can we guarantee the template will have a branch? It seems we're trusting the client to have good data if I read the expected call stack properly.

If we can't guarantee a valid branch name, would it make sense to default it to the old `master` default?